File: api/airTicket/src/main/java/com/xiang/airTicket/entity/AirPort.java
package com.xiang.airTicket.entity;

import com.fasterxml.jackson.annotation.JsonView;

import javax.persistence.*;

// 机场表
@Entity
public class AirPort {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // id

    private String icon; // 图标

    private String name;

    @ManyToOne
    @JsonView(CityJsonView.class)
    private City city; // 所属城市

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public City getCity() {
        return city;
    }

    public void setCity(City city) {
        this.city = city;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public interface CityJsonView {
    }
}

File: webapp/airTicket/src/app/entity/AirPort.ts
// 机场表

import {City} from './City';

export class AirPort {

  id: number; // id

  icon: string; // 图标

  name: string;

  city: City; // 所属城市

  constructor(name: string, icon: string, city: City) {
    this.icon = icon;
    this.name = name;
    this.city = city;
  }

}
